src/objective-c/GRPCClient/GRPCCall.m

```diff
@@ -106,6 +107,33 @@ @implementation GRPCCall {
 // TODO(jcanizales): If grpc_init is idempotent, this should be changed from load to initialize.
 + (void)load {
   grpc_init();
+  callFlags = [NSMutableDictionary dictionary];
+}
+
++ (void)setCallAttribute:(GRPCCallAttr)callAttr host:(NSString *)host path:(NSString *)path {
+  NSString *hostAndPath = [NSString stringWithFormat:@"%@%@", host, path];
+  switch (callAttr) {
+    case GRPCCallAttrDefault:
+      callFlags[hostAndPath] = @0;
+      break;
+    case GRPCCallAttrIdempotentRequest:
+      callFlags[hostAndPath] = @GRPC_INITIAL_METADATA_IDEMPOTENT_REQUEST;
+      break;
+    case GRPCCallAttrCacheableRequest:
+      callFlags[hostAndPath] = @GRPC_INITIAL_METADATA_CACHEABLE_REQUEST;
+      break;
+    default:
+      break;
+  }
+}
+
++ (uint32_t)getCallFlag:(NSString *)host path:(NSString *)path {
+  NSString *hostAndPath = [NSString stringWithFormat:@"%@%@", host, path];
+  if (nil != [callFlags objectForKey:hostAndPath]) {
+    return [callFlags[hostAndPath] intValue];
+  } else {
+    return 0;
+  }
```

Comment:
`return callFlags[hostAndPath].intValue;` gives you what you want.

`if (x)` evaluates to false if `x` is `nil`, so you never need to compare to `nil`. And you can always send ObjC messages to `nil` without worry; the return value will be `0`, `nil`, or `NULL`, depending on the return type. In this case, `[nil intValue] == 0`.